xbmc/network/linux/NetworkLinux.cpp

```diff
@@ -321,15 +331,36 @@ CNetworkLinux::~CNetworkLinux(void)
   if (m_sock != -1)
     close(CNetworkLinux::m_sock);
 
-  vector<CNetworkInterface*>::iterator it = m_interfaces.begin();
+  m_updThread->StopThread(false);
+  CSingleLock lock(m_lock);
+  InterfacesClear();
+  DeleteRemoved();
+  m_updThread->StopThread(true);
+}
+
+void CNetworkLinux::DeleteRemoved(void)
+{
+  std::vector<CNetworkInterface*>::iterator it = m_interfaces.begin();
   while(it != m_interfaces.end())
   {
+    if (!((CNetworkInterfaceLinux*)*it)->IsRemoved())
+    {
+      it++;
```

Comment:
might as well change this to pre-increment or we'll get performance warnings from cppcheck